process-document/src/test/java/com/ritense/processdocument/web/TaskListResourceIntTest.kt

```diff
@@ -0,0 +1,175 @@
+package com.ritense.processdocument.web
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.ritense.authorization.AuthorizationContext
+import com.ritense.case.domain.ColumnDefaultSort
+import com.ritense.case.domain.DisplayType
+import com.ritense.case.domain.EmptyDisplayTypeParameter
+import com.ritense.case.domain.TaskListColumn
+import com.ritense.case.domain.TaskListColumnId
+import com.ritense.case.repository.TaskListColumnRepository
+import com.ritense.document.domain.impl.request.NewDocumentRequest
+import com.ritense.processdocument.BaseIntegrationTest
+import com.ritense.processdocument.domain.impl.request.NewDocumentAndStartProcessRequest
+import com.ritense.processdocument.web.request.TaskListSearchDto
+import com.ritense.valtimo.contract.authentication.AuthoritiesConstants
+import com.ritense.valtimo.service.CamundaTaskService
+import java.nio.charset.StandardCharsets
+import org.assertj.core.api.Assertions.assertThat
+import org.camunda.bpm.engine.RuntimeService
+import org.camunda.bpm.engine.TaskService
+import org.hamcrest.Matchers.hasItems
+import org.hamcrest.Matchers.hasKey
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.http.MediaType
+import org.springframework.security.test.context.support.WithMockUser
+import org.springframework.test.web.servlet.MockMvc
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders
+import org.springframework.test.web.servlet.result.MockMvcResultHandlers
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers
+import org.springframework.test.web.servlet.setup.MockMvcBuilders
+import org.springframework.transaction.annotation.Transactional
+import org.springframework.web.context.WebApplicationContext
+
+class TaskListResourceIntTest : BaseIntegrationTest() {
+
+    lateinit var mockMvc: MockMvc
+
+    @Autowired
+    lateinit var objectMapper: ObjectMapper
+
+    @Autowired
+    lateinit var webApplicationContext: WebApplicationContext
+
+    @Autowired
+    lateinit var taskListColumnRepository: TaskListColumnRepository
+
+    @Autowired
+    lateinit var runtimeService: RuntimeService
+
+    @Autowired
+    lateinit var taskService: TaskService
+
+    @BeforeEach
+    fun setUp() {
+        mockMvc = MockMvcBuilders.webAppContextSetup(webApplicationContext).build()
+
+        val allProcessIds = runtimeService.createProcessInstanceQuery().list().map { it.id }
+        runtimeService.deleteProcessInstances(allProcessIds, "Cleaning for tests", true, false)
```

Comment:
Minor: We usually do this by adding `@Transactional` above the class. Spring will then make sure all database changes are rolled back after the test

Reply:
I've just tried this, but unfortunately my test break at the `assertThat(taskList).hasSize(testUserData.size)` when multiple tests are ran.